### PR TITLE
Added file extension checking for all data operations.

### PR DESCRIPTION
**Pull request by JosephDiPalma**
Filter files to avoid trying to process invalid images with invalid extensions.

### CODE REVIEW

File: code/utils.py [hunk 1]
```diff
@@ -124,11 +115,8 @@ def get_csv_paths(folder: Path) -> List[Path]:
     Returns:
         A list of the paths to the CSV files in the folder.
     """
-    return sorted([
-        folder.joinpath(f.name)
-        for f in folder.iterdir() if ((folder.joinpath(f.name).is_file()) and (
-            "csv" in f.name) and (".DS_Store" not in f.name))
-    ],
+    return sorted([folder.joinpath(f.name) for f in folder.iterdir() if (
+                (folder.joinpath(f.name).is_file()) and ("csv" in f.name) and (".DS_Store" not in f.name) and (f.suffix.casefold() in IMAGE_EXTS))],
```
Comment (ntomita) [inline code/utils.py:119]: This should not check image_exts, as targeted files are csv. recommend to revert changes

File: code/utils.py [hunk 2]
```diff
@@ -87,11 +81,8 @@ def get_subfolder_paths(folder: Path) -> List[Path]:
     Returns:
         A list containing the paths of the subfolders.
     """
-    return sorted([
-        folder.joinpath(f.name) for f in folder.iterdir()
-        if ((folder.joinpath(f.name).is_dir()) and (".DS_Store" not in f.name))
-    ],
-                  key=str)
+    return sorted([folder.joinpath(f.name) for f in folder.iterdir() if
+                   ((folder.joinpath(f.name).is_dir()) and (".DS_Store" not in f.name) and (f.suffix.casefold() in IMAGE_EXTS))], key=str)
```
Comment (ntomita) [inline code/utils.py:85]: This should not check extensions; this function is to list subfolders (see .is_dir()).

File: code/utils.py [hunk 3]
```diff
@@ -70,11 +67,8 @@ def get_image_paths(folder: Path) -> List[Path]:
     Returns:
         A list of the full paths to the images in the folder.
     """
-    return sorted([
-        folder.joinpath(f.name) for f in folder.iterdir() if ((
-            folder.joinpath(f.name).is_file()) and (".DS_Store" not in f.name))
-    ],
-                  key=str)
+    return sorted([folder.joinpath(f.name) for f in folder.iterdir() if
+                   ((folder.joinpath(f.name).is_file()) and (".DS_Store" not in f.name) and (f.suffix.casefold() in IMAGE_EXTS))], key=str)
```
Comment (ntomita) [inline code/utils.py:71]: checking ".DS_Store" is not necessary here as it doesn't have image_exts anyway.

File: code/utils.py [hunk 4]
```diff
@@ -53,11 +53,8 @@ def get_image_names(folder: Path) -> List[Path]:
     Returns:
         A list of the names with paths of the images in a folder.
     """
-    return sorted([
-        Path(f.name) for f in folder.iterdir() if ((
-            folder.joinpath(f.name).is_file()) and (".DS_Store" not in f.name))
-    ],
-                  key=str)
+    return sorted([Path(f.name) for f in folder.iterdir() if
+                   ((folder.joinpath(f.name).is_file()) and (".DS_Store" not in f.name) and (f.suffix.casefold() in IMAGE_EXTS))], key=str)
```
Comment (ntomita) [inline code/utils.py:57]: checking ".DS_Store" is not necessary

File: code/utils.py [hunk 5]
```diff
@@ -20,11 +23,8 @@ def get_classes(folder: Path) -> List[str]:
     Returns:
         A list of strings corresponding to the class names.
     """
-    return sorted([
-        f.name for f in folder.iterdir()
-        if ((folder.joinpath(f.name).is_dir()) and (".DS_Store" not in f.name))
-    ],
-                  key=str)
+    return sorted([f.name for f in folder.iterdir() if
+                   ((folder.joinpath(f.name).is_dir()) and (".DS_Store" not in f.name) and (f.suffix.casefold() in IMAGE_EXTS))], key=str)
```
Comment (ntomita) [inline code/utils.py:27]: as .is_dir() suggests, this function is about folders, so no need to check extension.

File: code/utils.py [hunk 6]
```diff
@@ -9,6 +9,9 @@
 from pathlib import Path
 from typing import (Dict, List)
 
+# Valid image extensions.
+IMAGE_EXTS = [".jpg", ".jpeg", ".png", ".svs", ".tif", ".tiff", ".png"]
```
Comment (ntomita) [inline code/utils.py:13]: duplicated ".png" item

**Review by ntomita — CHANGES_REQUESTED** [review]
Hi Joseph, thanks for this PR. Please see my comments!

File: code/utils.py (hunk 3 above)
Comment (JosephDiPalma) [reply code/utils.py:71]: We still should check in case valid images are found within the ".DS_Store" folder.

File: code/utils.py (hunk 4 above)
Comment (JosephDiPalma) [reply code/utils.py:57]: We still should check in case valid images are found within the ".DS_Store" folder.

File: code/utils.py (hunk 3 above)
Comment (ntomita) [reply code/utils.py:71]: It already checks if the entity is file, .is_file(), so I think that will not happen, what do you think?

Comment (JosephDiPalma) [reply code/utils.py:71]: I was thinking of a file with name "/home/person/data/.DS_Store/image_1.png".

File: code/utils.py (hunk 6 above)
Comment (ntomita) [reply code/utils.py:13]: Could you address this too?

**Pull request merged**